packages/gestalt/src/contexts/OnNavigation.js

```diff
@@ -0,0 +1,54 @@
+// @flow strict
+import React, { useContext, createContext, type Context, type Element, type Node } from 'react';
+import PropTypes from 'prop-types';
+
+type EventHandlerType = ({|
+  +event: SyntheticEvent<>,
+|}) => void;
+export type onNavigationOptionsType = {|
+  [string]: Node | EventHandlerType,
+|};
+
+export const onNavigationOptionsPropType: React$PropType$Primitive<{ +[string]: mixed }> =
```

Comment:
You should be able to use the flow type defined above here

Reply:
I already tried... It breaks later

Reply:
Maybe we can review types, later on, it works for now.